Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Surfaces/EyeSurface.azsli

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Atom/Features/PBR/Surfaces/AnisotropicSurfaceData.azsli>
+#include <Atom/Features/PBR/Surfaces/BasePbrSurfaceData.azsli>
+#include <Atom/Features/PBR/Surfaces/ClearCoatSurfaceData.azsli>
+#include <Atom/Features/PBR/Surfaces/TransmissionSurfaceData.azsli>
+
+class Surface
+{
+    TransmissionSurfaceData transmission;
+
+#if ENABLE_CLEAR_COAT
+    ClearCoatSurfaceData clearCoat;
+#endif
+
+    // ------- BasePbrSurfaceData -------
+    
+    float3 position;            //!< Position in world-space
+    float3 normal;              //!< Normal in world-space
+    float3 vertexNormal;        //!< Vertex normal in world-space
+    float3 albedo;              //!< Albedo color of the non-metallic material, will be multiplied against the diffuse lighting value
+    float3 specularF0;          //!< Fresnel f0 spectral value of the surface
+    float roughnessLinear;      //!< Perceptually linear roughness value authored by artists. Must be remapped to roughnessA before use
+    float roughnessA;           //!< Actual roughness value ( a.k.a. "alpha roughness") to be used in microfacet calculations
+    float roughnessA2;          //!< Alpha roughness ^ 2 (i.e. roughnessA * roughnessA), used in GGX, cached here for perfromance
+    float subsurfaceScatteringFactor;
+
+    //! Applies specular anti-aliasing to roughnessA2 
+    void ApplySpecularAA();
+
+    //! Calculates roughnessA and roughnessA2 after roughness has been set
+    void CalculateRoughnessA();
+
+    //! Sets albedo and specularF0 using metallic workflow
+    void SetAlbedoAndSpecularF0(float3 baseColor, float specularF0Factor);
+
+};
+
+
+// Specular Anti-Aliasing technique from this paper:
+// http://www.jp.square-enix.com/tech/library/pdf/ImprovedGeometricSpecularAA.pdf
```

Comment:
nice

Reply:
Actually this is not mine, it's just duplicated code which is present in most Surfaces azsli's :P